packages/zoe/src/contractSupport/zoeHelpers.js

```diff
@@ -206,6 +206,76 @@ export const makeZoeHelpers = zcf => {
         });
         zoeService.offer(invite);
       }),
+    /**
+     * Escrow a payment with Zoe and reallocate the amount of the
+     * payment to a recipient.
+     *
+     * @param {Object} obj
+     * @param {Payment} obj.payment
+     * @param {String} obj.keyword
+     * @param {Handle} obj.recipientHandle
+     * @returns {Promise<undefined>}
+     *
+     */
+    escrowAndAllocateTo: ({ payment, keyword, recipientHandle }) => {
+      // We will create a temporary offer to be able to escrow our payment
+      // with Zoe.
+      let tempHandle;
```

Comment:
I hate this kind of stateful coupling by assignment.

Put everything below `offer` into the offerHook. Then have the outer function return the `outcome` promise. Since the outcome promise is now a promise for `undefined`, it's clearer if the new offerHook function explicitly `return undefined;`.

At that point the offerHook function is large, and should be pulled out into a separately defined function.

Reply:
I'm not sure I understood exactly, but I tried to refactor according to what you said here. It's difficult because we need the invite to be redeemed before most of the code gets called, so we can't really move things into the offerHook which is specified before the invite redemption.